### PR TITLE
fix: more expressive errors

### LINKED ISSUE #22449: Backport to 2.0: better backup error message
Backports #22446

### PR DESCRIPTION
**Pull request by lesam**
Closes #22449

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: http/errors_test.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"encoding/json"
 	stderrors "errors"
+	"go.uber.org/zap/zaptest"
```
Comment (danxmoran) [inline http/errors_test.go:7]: Needs reformatted?

File: http/legacy/influx1x_authentication_handler_test.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package legacy
 import (
 	"context"
 	"fmt"
+	"go.uber.org/zap/zaptest"
```
Comment (danxmoran) [inline http/legacy/influx1x_authentication_handler_test.go:6]: Same here

File: http/legacy/write_handler_test.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package legacy
 import (
 	"context"
 	"fmt"
+	"go.uber.org/zap"
```
Comment (danxmoran) [inline http/legacy/write_handler_test.go:6]: And here

File: kit/transport/http/error_handler.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"go.uber.org/zap"
```
Comment (danxmoran) [inline kit/transport/http/error_handler.go:9]: And here

File: kit/transport/http/error_handler_test.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package http_test
 import (
 	"context"
 	"fmt"
+	"go.uber.org/zap/zaptest"
```
Comment (danxmoran) [inline kit/transport/http/error_handler_test.go:6]: Last one

**Review by danxmoran — APPROVED** [review]
LGTM overall but needs some import cleanup

**Comment by lesam** [comment]
Ideally we'd run the formatting we want in the linter... `gci` is pretty good.

**Pull request merged**